### DISCUSSION TITLE
can not see ref sequence

### DISCUSSION THREAD

**Discussion by cxbiosyn** [GMOD/jbrowse-components#4485] [Q&A] [answered]
Dear Jbrowse2 developer,

I got this error from JBrowse, here is the stack trace:

```
Error: problem decompressing block: incorrect gzip header check

../../../node_modules/@gmod/bgzf-filehandle/src/unzip-pako.ts:51:13 (at i ()
../../../node_modules/@gmod/bgzf-filehandle/src/bgzFilehandle.ts:89:34 (at d._readAndUncompressBlock ()
../../../node_modules/@gmod/bgzf-filehandle/src/bgzFilehandle.ts:112:13 (at async d.read ()
../../../node_modules/@gmod/indexedfasta/src/indexedFasta.ts:218:5 (at async l._fetchFromIndexEntry ()
JBrowse 2.11.2
```

https://xliugroup.com:39000/?session=local-moIaIGLQw2Ncf8C85Cpr8
<img width="1429" alt="image" src="https://github.com/user-attachments/assets/386c23c4-749e-46ad-866c-182938a68e72">

<img width="953" alt="image" src="https://github.com/user-attachments/assets/8f8c8b82-3513-4a92-bc61-472e1c43ca12">

The file I used:
I download genome file from NCBI, name it MG1363_NC_009004.1.fasta
Then I just changed the ".fasta" to ".fa"

in my server, I used samtools to covert MG1363_NC_009004.1.fa 
the command I used:
	 bgzip -i MG1363_NC_009004.1.fa
	 samtools faidx MG1363_NC_009004.1.fa.gz

Regards
Chenxi

**Comment by cmdcolin** [comment, marked as answer]
Hi there,
This is likely something weird with your server side setup. I don't know exactly what but if i for example download your fa.gz, fa.gz.gzi, and .fa.gz.fai files, then load it into my local instance, it works fine. this tends to mean your server is adding some weird HTTP headers to gzip files.

i can see you have something like a REST API serving the files from the URL containing question mark (https://xliugroup.com/shared-files/1380/?MG1363_NC_009004.1.fa.gz)

i think the api might not be responding to the Range header properly perhaps

**Reply by cmdcolin** [reply]
evidence for the not responding to Range header: the network tab on your website says the entire .fa.gz file is downloaded. the 698kb is the entire size of .fa.gz instead of a requested byte range of it
![Screenshot from 2024-07-18 08-49-52](https://github.com/user-attachments/assets/e0886e15-13eb-400a-bdb9-0185a901fdc0)

**Comment by cxbiosyn** [comment]
Thank you for letting me know it works at your local instance

there are indeed too many problems if I insist using url from my website
I try to do that because I wanted to make students without bioinformatic background directly use the web interface

But probably this will cost too much effort for me to debug, I will try still to working on this, at the same time, I tried to use the jbrowse cli, but having another problem

here is my config file
[config.json](https://github.com/user-attachments/files/16302108/config.json)

for normal page, it's working
https://xliugroup.com:39000/?session=local-moIaIGLQw2Ncf8C85Cpr8
<img width="1424" alt="image" src="https://github.com/user-attachments/assets/990e056d-b38c-4ddd-b1ce-f9d530259370">

but for admin page, somethings wrong 
<img width="1435" alt="image" src="https://github.com/user-attachments/assets/9ff23503-469d-49f1-b41f-e917f8044e77">

I don't understand...
